src/ppl/src/Slots.h

```diff
@@ -50,6 +50,7 @@ namespace ppl {
 class Interval
 {
  public:
+  Interval() = default;
```

Comment:
What was the warning that required this?

Reply:
The default constructor for the `Constraint` struct required this, but I noticed we don't need either of these default actors. I've updated the code in my last commit.